pkg/api/proxy.go

```diff
@@ -0,0 +1,110 @@
+package api
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/dchest/siphash"
+	"github.com/gorilla/mux"
+)
+
+func ClusterProxy(ctrlr *Controller) func(http.HandlerFunc) http.HandlerFunc {
+	return func(next http.HandlerFunc) http.HandlerFunc {
+		return http.HandlerFunc(func(response http.ResponseWriter, request *http.Request) {
+			config := ctrlr.Config
+			logger := ctrlr.Log
+
+			// if no cluster or single-node cluster, handle locally
+			if config.Cluster == nil || len(config.Cluster.Members) == 1 {
+				next.ServeHTTP(response, request)
+				return
+			}
+
+			vars := mux.Vars(request)
+			name, ok := vars["name"]
+
+			if !ok || name == "" {
+				response.WriteHeader(http.StatusNotFound)
+				return
+			}
+
+			h := siphash.New([]byte(config.Cluster.HashKey))
+			h.Write([]byte(name))
+			sum64 := h.Sum64()
+
+			targetMember := config.Cluster.Members[sum64%uint64(len(config.Cluster.Members))]
+
+			// from the member list and our DNS/IP address, figure out if this request should be handled locally
+			localMember := fmt.Sprintf("%s:%s", config.HTTP.Address, config.HTTP.Port)
+			if targetMember == localMember {
+				logger.Debug().Msg("Target cluster member is the local member. Handling request locally.")
```

Comment:
use lowercase log msgs so s/Target/target/
everywhere ...

Also add more details such as logger.Debug().Str("name", name).Msg() etc.

Reply:
Addressed the log message case comment.

Will look at adding more relevant details to the logger.

Reply:
Added "name" of the repo to the logger. Thanks for the suggestion!